validator/testcases/frameworks.py

```diff
@@ -0,0 +1,546 @@
+import json
+import re
+
+from validator.constants import LIBRARY_METADATA
+from validator.decorator import cached_property
+from validator.testcases import libraries
+from validator.version import Version
+
+
+def fake_version(files):
+    """
+    Returns a hash to act in place of a version number, given the contents
+    of a set of known files.
+    """
+    import hashlib
+    return hashlib.sha256("".join(files)).hexdigest()[:16]
+
+
+class MetaFramework(type):
+    def __new__(mcls, name, bases, dict_):
+        cls = super(MetaFramework, mcls).__new__(mcls, name, bases, dict_)
+        if cls.NAME:
```

Comment:
It seems this is only to not list the base `Framework` class, so maybe have a specific case just for it, `if cls.__name__ != 'Framework':`